python/lsst/ip/diffim/subtractImages.py

```diff
@@ -167,7 +172,59 @@ def __init__(self, **kwargs):
         self.convolutionControl.setDoNormalize(False)
         self.convolutionControl.setDoCopyEdge(True)
 
-    def run(self, template, science, sources):
+    @lsst.utils.inheritDoc(lsst.pipe.base.PipelineTask)
+    def runQuantum(self, butlerQC: lsst.pipe.base.ButlerQuantumContext,
+                   inputRefs: lsst.pipe.base.InputQuantizedConnection,
+                   outputRefs: lsst.pipe.base.OutputQuantizedConnection):
+        inputs = butlerQC.get(inputRefs)
+        self.log.info("Processing %s", butlerQC.quantum.dataId)
+
+        if self.config.doApplyFinalizedPsf:
+            finalizedPsfApCorrCatalog = inputs.get("finalizedPsfApCorrCatalog", None)
+            inputs["science"] = self.prepareCalibratedExposure(
+                inputs["science"],
+                finalizedPsfApCorrCatalog=finalizedPsfApCorrCatalog
+            )
+        subtractResults = self.run(**inputs)
+        return subtractResults
+
+    def prepareCalibratedExposure(self, exposure, finalizedPsfApCorrCatalog=None):
+        """Prepare a calibrated exposure and apply finalized psf if so configured.
+
+        Parameters
+        ----------
+        exposure : `lsst.afw.image.exposure.Exposure`
+            Input exposure to adjust calibrations.
+        finalizedPsfApCorrCatalog : `lsst.afw.table.ExposureCatalog`, optional
+            Exposure catalog with finalized psf models and aperture correction
+            maps to be applied if config.doApplyFinalizedPsf=True.  Catalog uses
+            the detector id for the catalog id, sorted on id for fast lookup.
+
+        Returns
+        -------
+        exposure : `lsst.afw.image.exposure.Exposure`
+            Exposure with adjusted calibrations.
+        """
+        detectorId = exposure.getInfo().getDetector().getId()
+
+        if finalizedPsfApCorrCatalog is not None:
+            row = finalizedPsfApCorrCatalog.find(detectorId)
+            if row is None:
+                self.log.warning("Detector id %s not found in finalizedPsfApCorrCatalog; "
```

Comment:
Should this be a `warning`, `error`, or `raise`?

Reply:
I'm going to leave these as `warning`s for now.